server/schedule/config/config.go

```diff
@@ -0,0 +1,95 @@
+package config
+
+import (
+	"sync"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/core/storelimit"
+)
+
+// RejectLeader is the label property type that suggests a store should not
+// have any region leaders.
+const RejectLeader = "reject-leader"
+
+var schedulerMap sync.Map
+
+// RegisterScheduler registers the scheduler type.
+func RegisterScheduler(typ string) {
+	schedulerMap.Store(typ, struct{}{})
+}
+
+// IsSchedulerRegistered checks if the named scheduler type is registered.
+func IsSchedulerRegistered(name string) bool {
+	_, ok := schedulerMap.Load(name)
+	return ok
+}
+
+// Config is the interface that wraps the Config related method.
+type Config interface {
```

Comment:
Each scheduler also has it's own config, How to distinguish them?

Reply:
A config instance that implements these interfaces can be used to start the scheduling services. And this interface is used for compatibility. As for the scheduler config, it doesn't rely on the server package so it can be defined without implementing this interface.

Reply:
maybe we can reduce the functions.